src/python/pants/bin/local_pants_runner.py

```diff
@@ -224,29 +224,30 @@ def run(self, start_time: float) -> ExitCode:
         self.run_tracker.start(run_start_time=start_time, specs=specs)
 
         with maybe_profiled(self.profile_path):
-            global_options = self.options.for_global_scope()
-
             if self.options.help_request:
                 return self._print_help(self.options.help_request)
 
+            global_options = self.options.for_global_scope()
+            goals = tuple(self.options.goals)
+            if not goals:
+                return PANTS_SUCCEEDED_EXIT_CODE
```

Comment:
No need to have this in the SWH block, as there should be no workunits in this case.